presto-orc/src/main/java/com/facebook/presto/orc/writer/DictionaryRowGroup.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.writer;
+
+import com.facebook.presto.orc.metadata.statistics.ColumnStatistics;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.Arrays;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.SizeOf.sizeOf;
+import static java.util.Objects.requireNonNull;
+
+class DictionaryRowGroup
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(DictionaryRowGroup.class).instanceSize();
+
+    private final byte[] byteIndexes;
+    private final short[] shortIndexes;
+    private final int[] intIndexes;
+    private final ColumnStatistics columnStatistics;
+    private final int valueCount;
+
+    public DictionaryRowGroup(int maxIndex, int[] dictionaryIndexes, int valueCount, ColumnStatistics columnStatistics)
+    {
+        requireNonNull(dictionaryIndexes, "dictionaryIndexes is null");
+        checkArgument(valueCount >= 0, "valueCount is negative");
+        requireNonNull(columnStatistics, "columnStatistics is null");
+
+        if (maxIndex <= Byte.MAX_VALUE) {
```

Comment:
What do you think about adding an enum indicating the type of stored indexes? It will improve the if-else-if-else in classes using DictionaryRowGroup.

Reply:
I am planning to make more changes here. The worst case, I have to deal with the dictionary is there are files, which only has 4K rows (one row group), but with schema like array<array<int>> and number of integers exceeding 10 million. I will make this segmented and more changes here, so will leave as it is for now.